gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClientV1Impl.java

```diff
@@ -179,26 +153,283 @@ public KubernetesClientV1Impl(Vertx vertx, KubernetesConfig kubernetesConfig) {
-                        ConfigMap configMap = response.bodyAsJsonObject().mapTo(ConfigMap.class);
-                        if (configMap == null) {
-                            LOGGER.error("Unable to retrieve configmap {} from namespace {}", configmapName, namespace);
+                        T data = null;
+                        if (resource.key == null) {
+                            data = response.bodyAsJsonObject().mapTo(type);
+                        } else {
+                            Object item = response.bodyAsJsonObject().mapTo(type);
+                            if (item instanceof ConfigMap) {
+                                data = (T) ((ConfigMap) item).getData().get(resource.key);
+                            } else if (item instanceof Secret) {
+                                data = (T) ((Secret) item).getData().get(resource.key);
+                            }
+                        }
+
+                        if (data == null) {
+                            LOGGER.error(
+                                "Unable to retrieve {} {} from namespace {}",
+                                resource.type.value,
+                                resource.name,
+                                resource.namespace
+                            );
                             return Maybe.empty();
                         } else {
-                            return Maybe.just(configMap);
+                            return Maybe.just(data);
                         }
                     }
                 }
             );
     }
 
+    @Override
+    public <T extends Event> Flowable<T> watch(String location, Class<T> type) {
+        KubeResource resource = parseLocation(location);
+        if (resource == null) {
+            return Flowable.empty();
+        }
+
+        String fieldSelector = resource.name == null ? "" : String.format("metadata.name=%s", resource.name);
+        Watch watch = watchMap.computeIfAbsent(location, Watch::new);
+
+        if (watchMap.get(watch.location).stopped) {
+            if (watchMap.get(watch.location).timerId != 0) {
+                LOGGER.info("Cancel timer for Kubernetes resource watcher {}.", watch.location);
+                vertx.cancelTimer(watchMap.get(watch.location).timerId);
+                watchMap.remove(watch.location);
+            }
+            return Flowable.empty();
+        }
+
+        LOGGER.info("Start watching namespace [{}] with fieldSelector [{}]", resource.namespace, fieldSelector);
+        return retrieveLastResourceVersion(resource.namespace, type)
+            .flatMapObservable(lrv -> fetchEvents(resource, lrv, fieldSelector, watch.location, type))
+            .doOnSubscribe(
+                disposable ->
+                    vertx.setPeriodic(
+                        config.getWebsocketTimeout(),
+                        l -> {
+                            watchMap.get(watch.location).timerId = l;
+                            watch(location, type);
```

Comment:
This is a note just to remind me that we have to discuss about this part because I think I'm missing something (watch returns a flowable but nothing subscribe on it 🧐)